diff --git a/arch/arm/arm-m/thread.c b/arch/arm/arm-m/thread.c
index 85ac4f03..34ebb86d 100644
--- a/arch/arm/arm-m/thread.c
+++ b/arch/arm/arm-m/thread.c
@@ -181,12 +181,14 @@ static void svc_handler(struct arm_cm_exception_frame_short *frame) {
             );
 		    break;
     case 1: {
+#ifdef WITH_AUX_HYPER_MODE        
         arch_disable_ints();
 	    void *(*curr_thx_func)(void) = app_info_header[6];
 	    void *curr_thx = curr_thx_func();
 	    struct thread *curr = get_current_thread();
 	    LTRACEF("%lx, %lx\n", curr_thx_func, curr_thx);
 	    arch_context_enter_pls(curr, curr_thx);
+#endif        
 	    break;
     }
 	}
diff --git a/arch/arm/include/arch/arch_ops.h b/arch/arm/include/arch/arch_ops.h
index df29d0c5..8da51a1f 100644
--- a/arch/arm/include/arch/arch_ops.h
+++ b/arch/arm/include/arch/arch_ops.h
@@ -13,7 +13,7 @@
 #include <kern/compiler.h>
 #include <kern/reg.h>
 #include <arch/arm.h>
-#ifndef WITH_AUX_HYPER_MODE
+#ifndef ARM_ONLY_THUMB
 #include <arch/arch_helpers.h>
 #endif
 #if ARM_ISA_ARMV7M
diff --git a/arch/include/arch.h b/arch/include/arch.h
index 155ad56f..680332dd 100644
--- a/arch/include/arch.h
+++ b/arch/include/arch.h
@@ -36,6 +36,6 @@ __END_CDECLS
 
 /* arch specific bits */
 #include <arch/defines.h>
-#ifndef WITH_AUX_HYPER_MODE
+#ifndef ARM_ONLY_THUMB
 #include <arch/arch_helpers.h>
 #endif
diff --git a/platform/stellaris/rules.mk b/platform/stellaris/rules.mk
index 4e7f0d3a..dbc3b871 100644
--- a/platform/stellaris/rules.mk
+++ b/platform/stellaris/rules.mk
@@ -31,6 +31,9 @@ ifeq ($(MEMSIZE),)
 $(error need to define MEMSIZE)
 endif
 
+KERNEL_ROMSIZE := 0x20000
+KERNEL_MEMSIZE := $(MEMSIZE)
+
 MODULE_SRCS += \
 	$(LOCAL_DIR)/debug.c \
 	$(LOCAL_DIR)/gpio.c \
diff --git a/project/lm3s6965evb-test.mk b/project/lm3s6965evb-test.mk
index acf25141..9dee8bfc 100644
--- a/project/lm3s6965evb-test.mk
+++ b/project/lm3s6965evb-test.mk
@@ -1,9 +1,13 @@
 MODULES += \
 
 MODULES += \
-	root/services/partition \
 	root/services/shell
 
+ifeq (true,$(call TOBOOL,$(ENABLE_MPU)))
+MODULES += \
+	root/services/partition
+endif
+
 include project/virtual/fs.mk
 # include project/virtual/test.mk
 include project/target/lm3s6965evb.mk
diff --git a/project/qemu-virt-arm32-r52-test.mk b/project/qemu-virt-arm32-r52-test.mk
index 420510ce..85c704fa 100644
--- a/project/qemu-virt-arm32-r52-test.mk
+++ b/project/qemu-virt-arm32-r52-test.mk
@@ -5,7 +5,6 @@ MODULES += \
 ifeq (true,$(call TOBOOL,$(WITH_AUX_HYPER_MODE)))
 MODULES += \
 	root/services/partition
-	
 endif
 
 ifeq (true,$(call TOBOOL,$(WITH_HYPER_MODE)))
